From b6d6800fc39538db50753df35d117b7b6a8c3311 Mon Sep 17 00:00:00 2001
From: Pauli Virtanen <pav@iki.fi>
Date: Sun, 25 Apr 2021 21:18:45 +0300
Subject: [PATCH 06/10] bluez5: don't unregister HFP/HSP profiles when shutting
 down

This apparently causes delays in shutdown under some conditions, and
closing the DBus connection should be enought to tell BlueZ that we are
shutting down.
---
 spa/plugins/bluez5/backend-native.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/spa/plugins/bluez5/backend-native.c b/spa/plugins/bluez5/backend-native.c
index ceb00c52..c546c2af 100644
--- a/spa/plugins/bluez5/backend-native.c
+++ b/spa/plugins/bluez5/backend-native.c
@@ -1870,8 +1870,6 @@ static int backend_native_free(void *data)
 
 	sco_close(backend);
 
-	backend_native_unregister_profiles(backend);
-
 #ifdef HAVE_BLUEZ_5_BACKEND_HSP_NATIVE
 	dbus_connection_unregister_object_path(backend->conn, PROFILE_HSP_AG);
 	dbus_connection_unregister_object_path(backend->conn, PROFILE_HSP_HS);
-- 
2.30.2

